tools/markdown_lint/linter.py

```
@@ -0,0 +1,826 @@
+        if actual_number != expected_number:
+            # Create a fix function that captures the correct expected number
+            def fix_ordered_number(line_content, expected=expected_number):
+                # Preserve the original indentation
+                match = re.match(r"^(\s*)\d+\.(.*)$", line_content)
+                if match:
+                    indent, rest = match.groups()
+                    return f"{indent}{expected}.{rest}"
+                return line_content
+
+            # Create a closure to capture the expected number
+            fix_func = lambda content: fix_ordered_number(content, expected_number)
```

Ruff (0.12.2)

Do not assign a `lambda` expression, use a `def`

Rewrite `fix_func` as a `def`

(E731)